### PR TITLE
word controller

### PR DESCRIPTION
**Pull request by NguyenTanDuc**

### CODE REVIEW

File: app/controllers/words_controller.rb [hunk 1]
```diff
@@ -1,4 +1,54 @@
...
+      flash[:success] = I18n.t :delete_success
+      redirect_to words_path
+    else
+      flash[:danger] = I18n.t :delete_fail
+      redirect_to words_path
+    end
+  end
+
+  private
+
```
Comment (vjnhat) [inline app/controllers/words_controller.rb]: remove this line

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -15,3 +15,17 @@
 //= require bootstrap
 //= require turbolinks
 //= require_tree .
+
+// ANSWERS CHECKBOX
+function checkone(stayChecked){
+  with(document.myForm)
+    {
+    for(i = 0; i < elements.length; i++)
```
Comment (vjnhat) [inline app/assets/javascripts/application.js]: Không được dùng for

File: app/views/words/index.html.erb [hunk 3]
```diff
@@ -2,14 +2,33 @@
...
-      <select class="col-md-3">
-        <option>Basic 500</option>
-        <option>Advance 200</option>
-        <option>Expert 100</option>
-      </select>        
-        <input value="1" type="radio" name="filter">learned
-        <input value="2" type="radio" name="filter">not learned
-        <input value="3" type="radio" name="filter">all              
+      <%= collection_select(:category, :category_id, Category.all, 
+                            :id, :name, {:prompt => false },
```
Comment (leduytungframgia) [inline app/views/words/index.html.erb]: delete space before }

File: app/views/shared/_error_messages.html.erb [hunk 4]
```diff
@@ -1,10 +1,9 @@
-<% if @user.errors.any? %>
-  <div id="error_explanation">
+<% if object.errors.any? %>
+  <div class="alert alert-danger">
+    You have <%= pluralize(object.errors.count, "errors") %> in this form.
```
Comment (brokenheart1712) [inline app/views/shared/_error_messages.html.erb]: No need to show number of errors. Just errors. Or you must implement I18n here. Don't do that.

File: app/views/words/index.html.erb [hunk 5]
```diff
@@ -17,22 +36,26 @@
...
       </tr>
+      <% @answers.each do |answer| %>
+        <tr>
+          <td><%= answer.word.id %></td>
+          <td><%= answer.word.content %></td>
+          <td><%= answer.content %></td>
+          <td>
+            <%= link_to "Edit", edit_word_path(answer.word.id) %>
+            <%= link_to "Delete", answer.word, method: :delete, 
+                                  data: { confirm: I18n.t(:delete_warning)},
```
Comment (leduytungframgia) [inline app/views/words/index.html.erb]: delete space

File: app/views/words/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<h1>Add new word</h1>
+<%= form_for @word, html: {name: "myForm"} do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
+  <table class="table table-bordered">
+    <tr>
+      <td>Categories</td>
+      <td>
+        <%= f.collection_select(:category_id, Category.all, 
+                                :id, :name, {:prompt => false },
```
Comment (leduytungframgia) [inline app/views/words/new.html.erb]: delete space

File: config/locales/en.yml [hunk 7]
```diff
@@ -27,3 +27,13 @@ en:
...
+  create_success: "Create success"
+  create_fail: "Create fail"
+  update_success: "Update success"
+  update_fail: "Update fail"
+  delete_success: "Delete success"
+  delete_fail: "Delete fail"
+  add_new_word: "Add new word"
+  edit_word: "Edit word"
+  delete_warning: "Are you sure?"
+
```
Comment (leduytungframgia) [inline config/locales/en.yml]: delete space

File: app/views/words/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    <tr>
+      <td><%= f.label :word %></td>
+      <td><%= f.text_field :content, class: "form-control" %></td>
+    </tr>
+    <%=  f.fields_for :answers do |builder|%>
+      <%= render "answer_fields", :f => builder %>
+    <% end %>    
+  </table>
+  <%= f.submit I18n.t(:edit_word), class: "btn btn-primary" %>
+<% end %>
```
Comment (brokenheart1712) [inline app/views/words/edit.html.erb]: Add space after *{* and before *}*

File: db/seeds.rb [hunk 9]
```diff
@@ -6,13 +6,20 @@
...
+  name = "Category#{n+1}"
+  Category.create!(name: name)
+end
+
+categories = Category.all
+30.times do |n|
   content = "English#{n+1}"
-  Word.create!(content: content)
+  categories.each {|category| category.words.create! content: content}
+  # Word.create!(content: content)
```
Comment (leduytungframgia) [inline db/seeds.rb]: delete remark

File: app/views/words/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    <tr>
+      <td><%= f.label :word %></td>
+      <td><%= f.text_field :content, class: "form-control" %></td>
+    </tr>
+    <%=  f.fields_for :answers do |builder|%>
+      <%= render "answer_fields", :f => builder %>
+    <% end %>    
+  </table>
+  <%= f.submit I18n.t(:add_new_word), class: "btn btn-primary" %>
+<% end %>
```
Comment (brokenheart1712) [inline app/views/words/new.html.erb]: Add space after *{* and before *}*

File: config/locales/en.yml (hunk 7 above)
Comment (brokenheart1712) [inline config/locales/en.yml]: Need a new line

File: app/controllers/words_controller.rb [hunk 11]
```diff
@@ -1,4 +1,53 @@
 class WordsController < ApplicationController
   def index
+    @answers = Word.answer_correct.paginate page: params[:page], per_page: 20
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: `Word.answer_correct.`???

File: app/models/word.rb [hunk 12]
```diff
@@ -1,6 +1,14 @@
...
   belongs_to :category
-  has_many :answers
-  has_many :lessons_words
+  has_many :answers, dependent: :destroy
+  has_many :lessons_words, dependent: :destroy
   has_many :lessons, through: :lessons_words
+
+  validates :content, presence: true, length: {minimum: 2, maximum: 20}
+   
+  accepts_nested_attributes_for :answers, :allow_destroy => true
```
Comment (dieunb) [inline app/models/word.rb]: coding convention

File: app/views/layouts/_header.html.erb [hunk 13]
```diff
@@ -7,13 +7,13 @@
           <%= link_to I18n.t(:home), root_path %>
```
Comment (dieunb) [inline app/views/layouts/_header.html.erb]: only use `t` instead of `I18n.t`

File: app/views/words/_answer_fields.html.erb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+<tr>
+  <td><%= f.label :content, 'Answer' %></td>
+  <td><%= f.text_field :content, class: "form-control" %></td>
+  <td><%= f.check_box :correct, onClick: "checkone(this);" %></td>
```
Comment (dieunb) [inline app/views/words/_answer_fields.html.erb]: don't using onClick

File: app/views/words/edit.html.erb [hunk 15]
```diff
@@ -0,0 +1,22 @@
+<h1>Add new word</h1>
+<%= form_for @word, html: {name: "wordForm"} do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
+  <table class="table table-bordered">
+    <tr>
+      <td>Categories</td>
+      <td>
+        <%= f.collection_select(:category_id, Category.all, 
+                                :id, :name, {:prompt => false},
```
Comment (dieunb) [inline app/views/words/edit.html.erb]: coding convention

File: app/views/words/edit.html.erb [hunk 16]
```diff
@@ -0,0 +1,22 @@
+<h1>Add new word</h1>
+<%= form_for @word, html: {name: "wordForm"} do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
+  <table class="table table-bordered">
+    <tr>
+      <td>Categories</td>
+      <td>
+        <%= f.collection_select(:category_id, Category.all, 
+                                :id, :name, {:prompt => false}, 
+                                {:class => "form-control"}) %>
```
Comment (dieunb) [inline app/views/words/edit.html.erb]: coding convention

File: app/views/words/edit.html.erb [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+      </td>
+    </tr>
+    <tr>
+      <td><%= f.label :word %></td>
+      <td><%= f.text_field :content, class: "form-control" %></td>
+    </tr>
+    <%=  f.fields_for :answers do |builder|%>
+      <%= render "answer_fields", :f => builder %>
```
Comment (dieunb) [inline app/views/words/edit.html.erb]: coding convention

File: app/views/words/index.html.erb [hunk 18]
```diff
@@ -2,14 +2,33 @@
...
-      <select class="col-md-3">
-        <option>Basic 500</option>
-        <option>Advance 200</option>
-        <option>Expert 100</option>
-      </select>        
-        <input value="1" type="radio" name="filter">learned
-        <input value="2" type="radio" name="filter">not learned
-        <input value="3" type="radio" name="filter">all              
+      <%= collection_select(:category, :category_id, Category.all, 
+                            :id, :name, {:prompt => false},
```
Comment (dieunb) [inline app/views/words/index.html.erb]: coding convention

File: app/views/words/index.html.erb [hunk 19]
```diff
@@ -2,14 +2,33 @@
...
-        <option>Basic 500</option>
-        <option>Advance 200</option>
-        <option>Expert 100</option>
-      </select>        
-        <input value="1" type="radio" name="filter">learned
-        <input value="2" type="radio" name="filter">not learned
-        <input value="3" type="radio" name="filter">all              
+      <%= collection_select(:category, :category_id, Category.all, 
+                            :id, :name, {:prompt => false}, 
+                            {:class => "form-control"}) %>
```
Comment (dieunb) [inline app/views/words/index.html.erb]: coding convention

File: app/views/words/new.html.erb [hunk 20]
```diff
@@ -0,0 +1,22 @@
+<h1>Add new word</h1>
+<%= form_for @word, html: {name: "wordForm"} do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
+  <table class="table table-bordered">
+    <tr>
+      <td>Categories</td>
+      <td>
+        <%= f.collection_select(:category_id, Category.all, 
+                                :id, :name, {:prompt => false},
```
Comment (dieunb) [inline app/views/words/new.html.erb]: coding convention

File: app/views/words/new.html.erb [hunk 21]
```diff
@@ -0,0 +1,22 @@
+<h1>Add new word</h1>
+<%= form_for @word, html: {name: "wordForm"} do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
+  <table class="table table-bordered">
+    <tr>
+      <td>Categories</td>
+      <td>
+        <%= f.collection_select(:category_id, Category.all, 
+                                :id, :name, {:prompt => false}, 
+                                {:class => "form-control"}) %>
```
Comment (dieunb) [inline app/views/words/new.html.erb]: coding convention

File: app/views/words/new.html.erb [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+      </td>
+    </tr>
+    <tr>
+      <td><%= f.label :word %></td>
+      <td><%= f.text_field :content, class: "form-control" %></td>
+    </tr>
+    <%=  f.fields_for :answers do |builder|%>
+      <%= render "answer_fields", :f => builder %>
```
Comment (dieunb) [inline app/views/words/new.html.erb]: coding convention

**Pull request merged**